ui/components/multichain/app-header/app-header-unlocked-content.tsx

```diff
@@ -139,15 +139,21 @@ export const AppHeaderUnlockedContent = ({
     origin &&
     origin !== browser.runtime.id;
 
-  const handleMainMenuOpened = () => {
-    trackEvent({
-      event: MetaMetricsEventName.NavMainMenuOpened,
-      category: MetaMetricsEventCategory.Navigation,
-      properties: {
-        location: 'Home',
-      },
+  const handleMainMenuToggle = () => {
+    setAccountOptionsMenuOpen((previous) => {
+      const isMenuOpen = !previous;
+      if (isMenuOpen) {
+        trackEvent({
```

Comment:
Why are we creating a MetaMetrics event in a React state change handler? Shouldn't we do that afterward?

Reply:
@mcmire That is existing prod code, tracking whenever the menu opens

Was trying to keep the focus of this PR about fixing the toggle

Reply:
Gotcha. Does this not work?

``` typescript
setAccountOptionsMenuOpen((previous) => !previous);
if (accountOptionsMenuOpen) {
  trackEvent({
    event: MetaMetricsEventName.NavMainMenuOpened,
    category: MetaMetricsEventCategory.Navigation,
    properties: {
      location: 'Home',
    },
  });
}
```

Or does it make sense to move the tracking into a `useEffect`?

Reply:
Calling 

```
if (accountOptionsMenuOpen) {
  trackEvent({
    event: MetaMetricsEventName.NavMainMenuOpened,
    category: MetaMetricsEventCategory.Navigation,
    properties: {
      location: 'Home',
    },
  });
}
```

directly like that will skew the tracking especially knowing the extension has existing excessive re-renders

useEffect might work but again susceptible to re-renders

IMO safest option without a full rewrite of this component and popover is the setState and logically close to the button click event

I mean we could also put evens on the Menu component itself when it renders but I think these are for another PR